vendor/bat-native-ledger/src/bat/ledger/internal/response/response_uphold.cc

```diff
@@ -0,0 +1,903 @@
+    return braveledger_uphold::UserStatus::PENDING;
+  }
+
+  if (status == "restricted") {
+    return braveledger_uphold::UserStatus::RESTRICTED;
+  }
+
+  if (status == "blocked") {
+    return braveledger_uphold::UserStatus::BLOCKED;
+  }
+
+  if (status == "ok") {
+    return braveledger_uphold::UserStatus::OK;
+  }
+
+  return braveledger_uphold::UserStatus::EMPTY;
+}
+
+}  // namespace
+
+namespace braveledger_response_util {
+
+// Request Url:
+// GET https://api.uphold.com/v0/me/cards/{wallet_address}
+//
+// Success:
+// OK (200)
+//
+// Response Format:
+// {
+//   "CreatedByApplicationId": "193a77cf-02e8-4e10-8127-8a1b5a8bfece",
+//   "address": {
+//     "wire": "XXXXXXXXXX"
+//   },
+//   "available": "0.00",
+//   "balance": "0.00",
+//   "currency": "BAT",
+//   "id": "bd91a720-f3f9-42f8-b2f5-19548004f6a7",
+//   "label": "Brave Browser",
+//   "lastTransactionAt": null,
+//   "settings": {
+//     "position": 1,
+//     "protected": false,
+//     "starred": true
+//   },
+//   "createdByApplicationClientId": "4c2b665ca060d912fec5c735c734859a06118cc8",
+//   "normalized": [
+//     {
+//       "available": "0.00",
+//       "balance": "0.00",
+//       "currency": "USD"
+//     }
+//   ],
+//   "wire": [
+//     {
+//       "accountName": "Uphold Europe Limited",
+//       "address": {
+//         "line1": "Tartu mnt 2",
+//         "line2": "10145 Tallinn, Estonia"
+//       },
+//       "bic": "LHVBEE22",
+//       "currency": "EUR",
+//       "iban": "EE76 7700 7710 0159 0178",
+//       "name": "AS LHV Pank"
+//     },
+//     {
+//       "accountName": "Uphold HQ, Inc.",
+//       "accountNumber": "XXXXXXXXXX",
+//       "address": {
+//         "line1": "1359 Broadway",
+//         "line2": "New York, NY 10018"
+//       },
+//       "bic": "MCBEUS33",
+//       "currency": "USD",
+//       "name": "Metropolitan Bank",
+//       "routingNumber": "XXXXXXXXX"
+//     }
+//   ]
+// }
+
+ledger::Result ParseFetchUpholdBalance(
+    const ledger::UrlResponse& response,
+    double* available) {
+  DCHECK(available);
+
+  if (response.status_code == net::HTTP_UNAUTHORIZED ||
+      response.status_code == net::HTTP_NOT_FOUND ||
+      response.status_code == net::HTTP_FORBIDDEN) {
+    return ledger::Result::EXPIRED_TOKEN;
+  }
+
+  if (response.status_code != net::HTTP_OK) {
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  base::Optional<base::Value> value = base::JSONReader::Read(response.body);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Invalid JSON");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    BLOG(0, "Invalid JSON");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  const auto* available_str = dictionary->FindStringKey("available");
+  if (!available_str) {
+    BLOG(0, "Missing available");
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  base::StringToDouble(*available_str, available);
```

Comment:
we need to handle return, see example above

Reply:
Fixed.